hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -608,90 +818,102 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
         """
-        cdef:
-            ExchangeBase taker_market = market_pair.taker.market
-            str taker_trading_pair = market_pair.taker.trading_pair
-            OrderBook taker_order_book = market_pair.taker.order_book
-            list buy_fill_records = self._order_fill_buy_events.get(market_pair, [])
-            list sell_fill_records = self._order_fill_sell_events.get(market_pair, [])
-            object buy_fill_quantity = sum([fill_event.amount for _, fill_event in buy_fill_records])
-            object sell_fill_quantity = sum([fill_event.amount for _, fill_event in sell_fill_records])
-            object taker_top
-            object hedged_order_quantity
-            object avg_fill_price
+        taker_trading_pair = market_pair.taker.trading_pair
+        buy_fill_records = self._order_fill_buy_events.get(market_pair, [])
+        sell_fill_records = self._order_fill_sell_events.get(market_pair, [])
+        buy_fill_quantity = sum([fill_event.amount for _, fill_event in buy_fill_records])
+        sell_fill_quantity = sum([fill_event.amount for _, fill_event in sell_fill_records])
 
         global s_decimal_zero
 
+        taker_market = market_pair.taker.market
+
+        # Convert maker order size (in maker base asset) to taker order size (in taker base asset)
+        _, _, quote_rate, _, _, base_rate, _, _, _ = self.get_conversion_rates()
+
         if buy_fill_quantity > 0:
+            # Maker buy
+            # Taker sell
+            taker_slippage_adjustment_factor = Decimal("1") - self._slippage_buffer
+
             hedged_order_quantity = min(
-                buy_fill_quantity,
-                taker_market.c_get_available_balance(market_pair.taker.base_asset) *
+                buy_fill_quantity * base_rate,
+                taker_market.get_available_balance(market_pair.taker.base_asset) *
                 self._order_size_taker_balance_factor
             )
-            quantized_hedge_amount = taker_market.c_quantize_order_amount(taker_trading_pair, Decimal(hedged_order_quantity))
-            taker_top = taker_market.c_get_price(taker_trading_pair, False)
+            quantized_hedge_amount = taker_market.quantize_order_amount(taker_trading_pair, Decimal(hedged_order_quantity))
+
             avg_fill_price = (sum([r.price * r.amount for _, r in buy_fill_records]) /
                               sum([r.amount for _, r in buy_fill_records]))
-            order_price = taker_market.c_get_price_for_volume(
-                taker_trading_pair, False, quantized_hedge_amount
-            ).result_price
+
+            maker_order_ids = [r.order_id for _, r in buy_fill_records]
+            maker_exchange_trade_ids = [r.exchange_trade_id for _, r in buy_fill_records]
+
+            if len(maker_order_ids) != 1 or len(maker_exchange_trade_ids) != 1:
+                self.logger().error("Multiple buy maker orders fills")
+                raise RuntimeError
```

Comment:
I'm not sure of what behaviour this will cause with the understanding that we don't want to handle multiple maker orders for now. What just resetting the flags so the bot can start a new cycle safely.

Reply:
A return stmt can even suffice.

Reply:
This has never happened so far and should not happen in fact, but it's there just in case it does happen by some accident to warn us about this anomaly so that we can investigate it further.